be/src/exec/pipeline/exchange/local_exchange_sink_operator.h

```diff
@@ -65,7 +65,7 @@ class LocalExchangeSinkOperator final : public Operator {
 private:
     bool _is_finished = false;
     const std::shared_ptr<LocalExchanger>& _exchanger;
-
+    RuntimeProfile::HighWaterMarkCounter* _peak_memory_usage_counter = nullptr;
```

Comment:
could we add it into the base class? I think other operator may need it too.

Reply:
I think it needn't. Most operators will not record its memory usage as LocalExchange does, and the memory hook will take care of them nicely.